### PR TITLE
Make rootless work under no_pivot_root

### PR DESCRIPTION
**Pull request by afbjorklund**
Previously only done for root isolation

### CODE REVIEW

**Comment by afbjorklund** [comment]
@giuseppe does this look OK ?

Same original issue as #1071

**Comment by giuseppe** [comment]
LGTM

File: cmd/buildah/unshare.go [hunk 1]
```diff
@@ -148,6 +148,14 @@ func maybeReexecUsingUserNamespace(c *cli.Context, evenForRoot bool) {
 		}
 	}
 
+	// Copy the no_pivot_root setting, for instance when running on tmpfs.
+	if no_pivot_root, present := os.LookupEnv("BUILDAH_NOPIVOT"); !present {
```
Comment (rhatdan) [inline cmd/buildah/unshare.go]: Validate check wants you to use noPivotRoot,  Does not like underscores for variables.

**Comment by afbjorklund** [comment]
Right. Think I managed to get the logic wrong, as well ("present")

Wonder why it worked ? Or maybe it didn't, tested the wrong thing ?

**Comment by afbjorklund** [comment]
Wonder if the unshare stuff was needed at all ? Does os.Environment() copy the existing one ?

File: cmd/buildah/unshare.go (hunk 1 above)
Comment (afbjorklund) [reply cmd/buildah/unshare.go]: Need to get "make validate" working here...

**Comment by afbjorklund** [comment]
Right, don't have to copy any variables - the Environment starts out as a copy  (isolation is an _override_)

File: run.go [hunk 2]
```diff
@@ -1130,10 +1130,16 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 	case IsolationChroot:
 		err = chroot.RunUsingChroot(spec, path, options.Stdin, options.Stdout, options.Stderr)
 	case IsolationOCIRootless:
+		var moreCreateArgs []string
```
Comment (rhatdan) [inline run.go]: Pain in the but nit:
var moreCreateArgs []string {"--no-new-keyring"}
if options.NoPivot {
			moreCreateArgs = append(moreCreateArgs, "--no-pivot")
}
Saves three lines.

Comment (afbjorklund) [reply run.go]: But adds a syntax error. See what you mean, though

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**